src/ai/backend/test/utils/exceptions.py

```diff
@@ -10,3 +10,10 @@ class UnexpectedSuccess(Exception):
 
     def __init__(self, message="Test succeeded unexpectedly."):
         super().__init__(message)
+
+
+class UnexpectedFailure(Exception):
```

Comment:
I don't see how this is different from just using an `AssertionError`.

In the case of `UnexpectedSuccess`, it needs to be specially marked, but in the opposite case, I think it's sufficient to either refine the exception handling or simply throw an AssertionException.

Reply:
`Assert` means that a specific condition should be met.
`AssertionError` means that the condition has been violated due to an unexpected reason, and as a result, some tests fail.

While AssertionError is indeed used in places outside of test code, there's no need to create a separate type like UnexpectedFailure just for that. On the contrary, I think our code should be modified to throw the appropriate error instead of throwing an AssertionError.